### PR TITLE
AMM refactor to reduce contract size

### PR DESCRIPTION
**Pull request by seafi**
* Moved view functions from `MinterAmm` into `AmmDataProvider`
* Set Light and AmmDataProvider addresses centrally instead of for each AMM
* Updated `SirenExchange` to use view functions from `AmmDataProvider`
* Consolidated AMM interfaces into a single `IMinterAmm`

Result: `MinterAmm` contract size reduced to 23.22

### CODE REVIEW

File: contracts/amm/AmmDataProvider.sol [hunk 1]
```diff
@@ -167,7 +169,7 @@ contract AmmDataProvider is IAmmDataProvider {
         uint256 bTokenAmount,
         uint256 collateralTokenBalance,
         uint256 bTokenPrice
-    ) external view override returns (uint256) {
+    ) public view override returns (uint256) {
```
Comment (MysticDakra) [inline contracts/amm/AmmDataProvider.sol:172]: We changed a few of these from external to public was there a reason for this? External I believe is more optimized for gas usage according to this article [https://gus-tavo-guim.medium.com/public-vs-external-functions-in-solidity-b46bcf0ba3ac](https://gus-tavo-guim.medium.com/public-vs-external-functions-in-solidity-b46bcf0ba3ac).

**Comment by MysticDakra** [comment]
We can remove the test `/test/amm/ammDataProviderUpdate.ts` as we removed this function entirely. We may want to add a test for the addressesProvider in the future but for this PR we should be okay.

File: contracts/amm/AmmDataProvider.sol [hunk 2]
```diff
@@ -405,7 +407,7 @@ contract AmmDataProvider is IAmmDataProvider {
     /// representing the price as a fraction of 1 collateral token unit
     /// @dev This function assumes that it will only be called on an OPEN Series; if the
     /// Series is EXPIRED, then the expirationDate - block.timestamp will throw an underflow error
-    function getPriceForExpiredSeries(uint64 seriesId, uint256 annualVolatility)
+    function getPriceForSeries(uint64 seriesId, uint256 annualVolatility)
```
Comment (siren-tamer) [inline contracts/amm/AmmDataProvider.sol:410]: Was this just a mis-named function before?

Comment (MysticDakra) [reply contracts/amm/AmmDataProvider.sol:410]: Yes it was getting price of Series not expired series.

**Review by siren-tamer — COMMENTED** [review]
One minor comment/question...

File: contracts/amm/AmmDataProvider.sol (hunk 2 above)
Comment (siren-tamer) [reply contracts/amm/AmmDataProvider.sol:410]: OK cool...

**Review by siren-tamer — APPROVED** [review]

**Review by MysticDakra — APPROVED** [review]

**Pull request merged**